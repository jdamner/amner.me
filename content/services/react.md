---
title: Javascript & React
---

I've worked with [React](https://reactjs.org) and [React Native](https://reactnative.dev) for a number of years and find the stateful nature of the React framework to really work well in my head. I love the component based approach and the ability to create reusable components that can be used in multiple places.

This site is build using React and the [NextJS](https://www.nextjs.org/) framework. I parse [markdown](https://daringfireball.net/projects/markdown/) files for the content. This site is totally open source too, so if you're curious about how it's built, you can check out the [source code](https://www.github.com/jdamner/amner.me).

Also, as a statically generated site, it's super fast and easy to deploy.
