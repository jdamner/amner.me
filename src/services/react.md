---
title: React
thumbnail: ../assets/react-logo.png
---

Working with SQL in raw queries, and using frameworks such as Eloquent and WP_Query.
Coaching and supporting a team of developers grow their skills and meet their objectives
Supporting Project Managers and Business Development teams for technical guidance on quoting and supporting project lifecycles.
Debugging and maintaining PHP projects built in many frameworks, including bespoke PHP frameworks developed in house.
Developing systems that interact with complex data, such as data provided via third-party tools and services that run on device and output in raw formats.