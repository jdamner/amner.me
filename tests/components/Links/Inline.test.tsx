import { render } from "@testing-library/react"
import InlineLink from "../../../src/components/Links/Inline"

describe("InlineLink", () => {
  it("renders correctly", () => {
    const { container } = render(<InlineLink href='/'>Test</InlineLink>)
    expect(container).toMatchSnapshot()
  })
})
