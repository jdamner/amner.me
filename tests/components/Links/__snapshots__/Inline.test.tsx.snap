// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineLink renders correctly 1`] = `
<div>
  <a
    class="underline hover:text-slate-800 hover:no-underline"
    href="/"
  >
    Test
  </a>
</div>
`;
